.devops/deploy-pipelines.yml

```diff
@@ -329,3 +329,78 @@ stages:
+          - script: |
+              # Give time to the scheduler to rest and reprocess old events
+              sleep 200 
+              
+              # Get the first event from the Azure Storage Queue
+              eventFromQueue_1=$(az storage message get \
+                --connection-string "$(SCHEDULER_CDC_QUEUE_CONNECTION_STRING)" \
+                --queue-name pagopa-u-itn-pay-wallet-cdc-queue \
+                --output tsv --query '[0].content')
+              
+              # Get the second event from the Azure Storage Queue
+              eventFromQueue_2=$(az storage message get \
+                --connection-string "$(SCHEDULER_CDC_QUEUE_CONNECTION_STRING)" \
+                --queue-name pagopa-u-itn-pay-wallet-cdc-queue \
+                --output tsv --query '[0].content')
+              
+              # Extract data from both events
+              eventDataFromQueue_1=$(echo "$eventFromQueue_1" | jq -c '.data')
+              eventDataFromQueue_2=$(echo "$eventFromQueue_2" | jq -c '.data')
+              
+              # Use IDs to determine which event is which
+              if [[ "$eventFromQueue_1" == *"00a06383-1495-4b90-88f6-80c5fecf554a"* ]]; then
+                eventDataFromQueueCards="$eventDataFromQueue_1"
+                eventDataFromQueuePaypal="$eventDataFromQueue_2"
+              elif [[ "$eventFromQueue_1" == *"a0a27446-e6d8-47d2-8786-ca393f19c30b"* ]]; then
+                eventDataFromQueuePaypal="$eventDataFromQueue_1"
+                eventDataFromQueueCards="$eventDataFromQueue_2"
+              else
+                echo "Unexpected event ID in eventFromQueue_1"
+                exit 1
+              fi
+              
+              expectedDataEventCards=$(cat automated-tests/onboarding-wallet-card.json | jq -c .)
+              expectedDataEventPaypal=$(cat automated-tests/onboarding-wallet-paypal.json | jq -c .)
+              
+              echo "EventDataFromQueueCards: $eventDataFromQueueCards"
+              echo "ExpectedEventDataCards: $expectedDataEventCards"
+              
+              echo "EventDataFromQueuePaypal: $eventDataFromQueuePaypal"
+              echo "ExpectedEventDataPaypal: $expectedDataEventPaypal"
+              
+              # Compare the events Cards
+              if [ "$eventDataFromQueueCards" == "$expectedDataEventCards" ]; then
+                echo "CARDS: test passed successfully"
+              else
+                echo "CARDS: fail! unexpected event"
+                exit 1
+              fi
+              
+              # Compare the events Paypal
+              if [ "$eventDataFromQueuePaypal" == "$expectedDataEventPaypal" ]; then
+                echo "PAYPAL: test passed successfully"
+              else
+                echo "PAYPAL: fail! unexpected event"
+                exit 1
+              fi
+            displayName: 'Get event from Azure Storage Queue and validate'
+
+  # --- END validation of CDC UAT --- #
```

Comment:
beyond the code, is the goal to have these tests blocking for the green deployment in uat?

Reply:
This is the current graph of the pipeline. As now, tests will not block green deployment, but nothing stops us from adding a new dependency (red line) to make sure both tests passed and green deployment approval has been granted before we proceed
![image](https://github.com/user-attachments/assets/16f957be-df8e-480c-a5bb-1e354bb9b7bb)

Reply:
I would not make tests mandatory for deployment just because tests could fails for reasons unrelated to the development (environment not available f.e.).
Leaving it as a parallel independent task would allow to proceed with deployment even if tests are failing if deployment administrators decide to do so.
We could also make deployment stage dependent from test ones but there should be an override mechanism to unlock a deployment pipeline

Reply:
I tests can fail for reasons unrelated to the development, i guess is probably better to keep them as they are currently (not mandatory)

Reply:
but before choosing whether to go with the deploy, are we forced to wait for the test results?
Maybe to have  the tests in parallel with the deployment in UAT will make these tests lose their value

Reply:
i don't think we are forced but it could be wise to wait for the test to be completed so that potential errors are spotted accordingly.

Reply:
can you kindly provide a diagram of the pipeline?

Reply:
a diagram like https://pagopa.atlassian.net/wiki/spaces/I/pages/729186305/ecommerce-payment-requests-service#deploy

Reply:
I think this pipeline should work as follows
![image](https://github.com/user-attachments/assets/9d0adb6c-f6a3-46fe-9c2b-8b7edcf5f675)

Red line is the new dependency. We must have a manual trigger to do green deploy and in parallel we have the tests

An admin can decide to wait for the test results or go ahead and deploy anyway

Reply:
Proposal of new scheme for deploy pipeline here
https://pagopa.atlassian.net/wiki/spaces/I/pages/1410203653/payment-wallet-scheduler-service

Reply:
good job!